File: Python3/771.Jewels_and_Stones.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        jewels_set = set()
        total = 0;
        for char in J:
            jewels_set.add(char)
        for stone in S:
            if stone in jewels_set:
                total += 1
        return total

File: Java/771.Jewels_and_Stones.java
import java.util.HashSet;

class Solution {
    public int numJewelsInStones(String J, String S) {
        Set<Character> jewelSet = new HashSet<>();
        int total = 0;
        for (int i = 0; i < J.length(); ++i) {
            jewelSet.add(J.charAt(i));
        }
        for (int j = 0; j < S.length(); ++j) {
            if (jewelSet.contains(S.charAt(j))) {
                ++total;
            }
        }
        return total;
    }
}